modules/substrate/src/lib.rs

```diff
@@ -61,25 +59,36 @@ pub(crate) type BridgedBlockHash<T> = HashOf<<T as Trait>::BridgedChain>;
 /// Header of the bridged chain.
 pub(crate) type BridgedHeader<T> = HeaderOf<<T as Trait>::BridgedChain>;
 
+pub trait Trait: frame_system::Trait {
+	/// Chain that we are bridging here.
+	type BridgedChain: Chain;
+}
+
 decl_storage! {
 	trait Store for Module<T: Trait> as SubstrateBridge {
+		/// The number of the highest block(s) we know of.
+		ChainTipHeight: BridgedBlockNumber<T>;
 		/// Hash of the header at the highest known height.
-		BestHeader: BridgedBlockHash<T>;
+		///
+		/// If there are multiple headers at the same "best" height
+		/// this will contain all of their hashes.
+		BestHeaders: Vec<BridgedBlockHash<T>>;
```

Comment:
Not necessarily a part of this PR, but if we want pruning this should probably be:
```rust
HeadersAtHeight: map hasher(identity) BridgedBlockNumber<T> => Vec<BridgedBlockHash<T>>;
```

instead of `BestHeaders`. Or do you have other ideas on how to organize pruning?

Reply:
I was kicking the pruning can down the road, which is why I opened a separate issue for it, lol

Reply:
What I mean is that when you'll start work on pruning, you may need to change this again (at least the way how pruning is organized in sibling pallet needs this change). So probably worth to think about pruning now. But it is up to you.

Reply:
Pruning will likely be the next thing I work on so I'm not concerned about making changes now